tensorboard/webapp/plugins/npmi/util/sort_annotations.ts

```diff
@@ -15,27 +15,29 @@ limitations under the License.
 import {
   AnnotationDataListing,
   AnnotationSort,
-  EmbeddingListing,
+  EmbeddingDataSet,
   SortOrder,
 } from '../store/npmi_types';
 import {stripMetricString} from './metric_type';
 
 export function sortAnnotations(
   annotationData: AnnotationDataListing,
   sort: AnnotationSort,
-  embeddingData: EmbeddingListing
+  embeddingData: EmbeddingDataSet | undefined
 ): string[] {
   const result = Object.keys(annotationData);
   const similarityBased =
     sort.order === SortOrder.DISSIMILAR || sort.order === SortOrder.SIMILAR;
   if (
     sort.metric === '' ||
-    (embeddingData[sort.metric] === undefined && similarityBased)
+    ((embeddingData === undefined ||
+      embeddingData.points[sort.metric] === undefined) &&
+      similarityBased)
   ) {
     return result;
   }
   const distanceData: {[annotation: string]: number} = similarityBased
-    ? calculateDistances(result, embeddingData, sort)
+    ? calculateDistances(result, embeddingData!, sort)
```

Comment:
Hmm with L33, there should not be a case when `embeddingData` is undefined.

Reply:
I don't get it I think. I get an error that this can be undefined if I remove the `!`. Don't know if that is what you mean, though.